packages/react/src/components/CodeSnippet/CodeSnippet.js

```diff
@@ -6,223 +6,172 @@
+import React, { useState, useRef, useLayoutEffect } from 'react';
 import classNames from 'classnames';
 import { ChevronDown16 } from '@carbon/icons-react';
 import { settings } from 'carbon-components';
 import Copy from '../Copy';
+import Button from '../Button';
 import CopyButton from '../CopyButton';
-import uid from '../../tools/uniqueId';
+import getUniqueId from '../../tools/uniqueId';
 
 const { prefix } = settings;
 
-export default class CodeSnippet extends Component {
-  static propTypes = {
-    /**
-     * Provide the type of Code Snippet
-     */
-    type: PropTypes.oneOf(['single', 'inline', 'multi']),
-
-    /**
-     * Specify an optional className to be applied to the container node
-     */
-    className: PropTypes.string,
-
-    /**
-     * Provide the content of your CodeSnippet as a string
-     */
-    children: PropTypes.string,
-
-    /**
-     * Specify the string displayed when the snippet is copied
-     */
-    feedback: PropTypes.string,
-
-    /**
-     * Specify the description for the Copy Button
-     */
-    copyButtonDescription: PropTypes.string,
-
-    /**
-     * An optional handler to listen to the `onClick` even fired by the Copy
-     * Button
-     */
-    onClick: PropTypes.func,
-
-    /**
-     * Specify a label to be read by screen readers on the containing <textbox>
-     * node
-     */
-    copyLabel: PropTypes.string,
-
-    /**
-     * Specify a label to be read by screen readers on the containing <textbox>
-     * node
-     */
-    ariaLabel: PropTypes.string,
-
-    /**
-     * Specify a string that is displayed when the Code Snippet text is more
-     * than 15 lines
-     */
-    showMoreText: PropTypes.string,
-
-    /**
-     * Specify a string that is displayed when the Code Snippet has been
-     * interacted with to show more lines
-     */
-    showLessText: PropTypes.string,
-
-    /**
-     * Specify whether you are using the light variant of the Code Snippet,
-     * typically used for inline snippet to display an alternate color
-     */
-    light: PropTypes.bool,
-  };
-
-  static defaultProps = {
-    type: 'single',
-    showMoreText: 'Show more',
-    showLessText: 'Show less',
-  };
-
-  state = {
-    shouldShowMoreLessBtn: false,
-    expandedCode: false,
-  };
-
-  componentDidMount() {
-    if (this.codeContent) {
-      if (this.codeContent.getBoundingClientRect().height > 255) {
-        this.setState({ shouldShowMoreLessBtn: true });
-      }
+const CodeSnippet = ({
+  className,
+  type,
+  children,
+  feedback,
+  onClick,
+  ariaLabel,
+  copyLabel, //TODO: Merge this prop to `ariaLabel` in `v11`
+  copyButtonDescription,
+  light,
+  showMoreText,
+  showLessText,
+  ...other
+}) => {
+  const [shouldShowMoreLessBtn, setShouldShowMoreLessBtn] = useState(false);
+  const [expandedCode, setExpandedCode] = useState(false);
+  const { current: codeContentRef } = useRef(null);
+  const { currenet: uid } = useRef(getUniqueId());
```

Comment:
this should be `current` right?

```suggestion
  const { current: uid } = useRef(getUniqueId());
```